### PR TITLE
Rework benchmarks and tests

### PR DESCRIPTION
**Pull request by thomasfaingnaert**
- Use same set of GEMM configurations for the test suite and the benchmark suite.
- Use CUPTI for benchmarking.

### CODE REVIEW

**Comment by maleadt** [comment]
Benchmark results for commit 9ad57ed3b4ed541cc799a8ade25592733c4665cf (comparing to 396effce3057ba8c048761ff4fc8ef200c5497b8):

| test | master | PR | Δmin |
|------|--------|----|------|
| `WMMA GEMM Float16*Float16=Float32 (256×256) · (256×256) (TT) OP (16, 16, 16)` | 178.0 μs ± 6.29% | 139.0 μs ± 5.05% | -18.9% :white_check_mark: |
| `Tropical GEMM Float32*Float32=Float32 (256×128) · (128×256) (NT) OP (8, 16, 2)` | 143.0 μs ± 4.18% | 194.0 μs ± 88.0% | +21.0% :x: |
| `FPU GEMM Float32*Float32=Float32 (128×128) · (128×128) (TN) OP (8, 16, 2)` | 175.0 μs ± 18.6% | 207.0 μs ± 7.31% | +42.6% :x: |
| `Tropical GEMM Float32*Float32=Float32 (128×128) · (128×128) (TN) OP (8, 16, 2)` | 136.0 μs ± 3.43% | 123.0 μs ± 4.33% | -11.6% :white_check_mark: |
| `WMMA GEMM Float16*Float16=Float32 (256×128) · (128×256) (TN) OP (16, 16, 16)` | 113.0 μs ± 7.75% | 152.0 μs ± 6.43% | +38.1% :x: |
| `WMMA GEMM Float16*Float16=Float16 (512×256) · (256×512) (NN) OP (16, 16, 16)` | 204.0 μs ± 5.71% | 282.0 μs ± 27.6% | +14.2% :x: |
| `Tropical GEMM Float32*Float32=Float32 (128×128) · (128×128) (NN) OP (8, 16, 2)` | 121.0 μs ± 5.5% | 107.0 μs ± 2.64% | -9.6% :white_check_mark: |
| `WMMA GEMM Float16*Float16=Float32 (256×256) · (256×256) (NN) OP (16, 16, 16)` | 152.0 μs ± 5.88% | 139.0 μs ± 6.46% | -8.3% :white_check_mark: |
| `FPU GEMM Float64*Float64=Float64 (128×128) · (128×128) (NT) OP (8, 16, 2)` | 1.48 ms ± 3.94% | 1.5 ms ± 0.101% | +13.0% :x: |
| `Tropical GEMM Float32*Float32=Float32 (128×256) · (256×128) (NT) OP (8, 16, 2)` | 215.0 μs ± 2.79% | 201.0 μs ± 1.0% | -5.4% :white_check_mark: |
| `FPU GEMM Float64*Float64=Float64 (128×128) · (128×128) (TN) OP (8, 16, 2)` | 1.44 ms ± 5.75% | 1.5 ms ± 0.127% | +12.7% :x: |
| `Tropical GEMM Float32*Float32=Float32 (128×128) · (128×128) (TT) OP (8, 16, 2)` | 296.0 μs ± 2.73% | 166.0 μs ± 2.65% | -43.0% :white_check_mark: |
| `Tropical GEMM Float32*Float32=Float32 (256×256) · (256×256) (TT) OP (8, 16, 2)` | 180.0 μs ± 1.73% | 284.0 μs ± 1.45% | +57.5% :x: |
| `FPU GEMM Int32*Int32=Int32 (128×128) · (128×128) (TT) OP (8, 16, 2)` | 143.0 μs ± 7.88% | 123.0 μs ± 8.77% | -12.7% :white_check_mark: |
| `Tropical GEMM Float32*Float32=Float32 (128×128) · (128×128) (NT) OP (8, 16, 2)` | 124.0 μs ± 15.4% | 134.0 μs ± 4.16% | +7.7% :x: |
| `Tropical GEMM Float32*Float32=Float32 (256×128) · (128×256) (TT) OP (8, 16, 2)` | 120.0 μs ± 8.43% | 194.0 μs ± 2.53% | +69.1% :x: |
| `FPU GEMM Float32*Float32=Float32 (128×128) · (128×128) (TT) OP (8, 16, 2)` | 182.0 μs ± 7.33% | 170.0 μs ± 7.75% | -6.4% :white_check_mark: |
| `FPU GEMM Float64*Float64=Float64 (256×256) · (256×256) (TN) OP (8, 16, 2)` | 2.88 ms ± 1.09% | 2.73 ms ± 6.32% | -9.7% :white_check_mark: |
| `WMMA GEMM Float16*Float16=Float32 (128×256) · (256×128) (TT) OP (16, 16, 16)` | 137.0 μs ± 4.67% | 155.0 μs ± 9.48% | +11.4% :x: |
| `WMMA GEMM Float16*Float16=Float32 (128×128) · (128×128) (NN) OP (16, 16, 16)` | 149.0 μs ± 5.8% | 108.0 μs ± 7.39% | -26.9% :white_check_mark: |
| `FPU GEMM Int16*Int16=Int16 (128×128) · (128×128) (TT) OP (8, 16, 2)` | 173.0 μs ± 9.21% | 130.0 μs ± 9.94% | +5.3% :x: |
| `Tropical GEMM Float32*Float32=Float32 (256×256) · (256×256) (TN) OP (8, 16, 2)` | 178.0 μs ± 4.1% | 164.0 μs ± 0.995% | -5.9% :white_check_mark: |
| `Tropical GEMM Float32*Float32=Float32 (256×256) · (256×256) (NN) OP (8, 16, 2)` | 175.0 μs ± 1.84% | 233.0 μs ± 2.31% | +31.3% :x: |
| `FPU GEMM Float32*Float32=Float64 (128×128) · (128×128) (TT) OP (8, 16, 2)` | 1.7 ms ± 5.84% | 1.77 ms ± 0.342% | +13.4% :x: |
| `FPU GEMM Float32*Float32=Float32 (128×128) · (128×128) (TN) OP (4, 16, 1)` | 192.0 μs ± 9.21% | 185.0 μs ± 7.39% | +6.7% :x: |
| `FPU GEMM Float16*Float16=Float32 (128×128) · (128×128) (TT) OP (8, 16, 2)` | 164.0 μs ± 15.8% | 111.0 μs ± 8.63% | -31.4% :white_check_mark: |
| `Tropical GEMM Float32*Float32=Float32 (256×128) · (128×256) (NN) OP (8, 16, 2)` | 197.0 μs ± 5.25% | 261.0 μs ± 1.36% | +37.2% :x: |
| `FPU GEMM Int32*Int32=Int32 (128×128) · (128×128) (TN) OP (8, 16, 2)` | 118.0 μs ± 8.62% | 129.0 μs ± 8.48% | +8.1% :x: |
| `FPU GEMM Float32*Float32=Float32 (128×128) · (128×128) (TN) OP (4, 8, 2)` | 135.0 μs ± 7.45% | 161.0 μs ± 7.44% | +18.5% :x: |
| `Tropical GEMM Float32*Float32=Float32 (128×256) · (256×128) (NN) OP (8, 16, 2)` | 223.0 μs ± 22.1% | 174.0 μs ± 0.731% | -11.1% :white_check_mark: |
| `WMMA GEMM Float16*Float16=Float32 (128×128) · (128×128) (NT) OP (16, 16, 16)` | 403.0 μs ± 732.0% | 150.0 μs ± 6.22% | +40.2% :x: |
| `FPU GEMM Int16*Int16=Int16 (128×128) · (128×128) (NT) OP (8, 16, 2)` | 125.0 μs ± 6.3% | 140.0 μs ± 9.06% | +9.2% :x: |
| `WMMA GEMM Float16*Float16=Float32 (128×256) · (256×128) (NN) OP (16, 16, 16)` | 169.0 μs ± 6.51% | 156.0 μs ± 6.23% | -5.1% :white_check_mark: |
| `FPU GEMM Float32*Float32=Float32 (128×128) · (128×128) (TT) OP (4, 16, 1)` | 175.0 μs ± 6.56% | 211.0 μs ± 8.16% | +16.1% :x: |
| `FPU GEMM Float32*Float32=Float64 (128×128) · (128×128) (NN) OP (8, 16, 2)` | 1.82 ms ± 0.121% | 1.79 ms ± 3.94% | -11.5% :white_check_mark: |
| `WMMA GEMM Float16*Float16=Float16 (256×256) · (256×256) (NT) OP (16, 16, 16)` | 201.0 μs ± 4.15% | 272.0 μs ± 20.0% | +8.2% :x: |
| `WMMA Complex GEMM Float16*Float16=Float32 (128×128) · (128×128) (NT) OP (16, 16, 16)` | 98.4 μs ± 2.96% | 110.0 μs ± 4.42% | +6.4% :x: |
| `FPU GEMM Float32*Float32=Float32 (128×128) · (128×128) (NN) OP (4, 16, 1)` | 176.0 μs ± 7.12% | 230.0 μs ± 8.02% | +29.4% :x: |
| `WMMA GEMM Float16*Float16=Float32 (256×128) · (128×256) (NN) OP (16, 16, 16)` | 149.0 μs ± 6.3% | 109.0 μs ± 7.09% | -26.8% :white_check_mark: |
| `WMMA GEMM Float16*Float16=Float16 (512×256) · (256×512) (NT) OP (16, 16, 16)` | 324.0 μs ± 19.7% | 207.0 μs ± 3.68% | -23.6% :white_check_mark: |

File: Manifest.toml [hunk 1]
```diff
@@ -1,20 +1,29 @@
 # This file is machine-generated - editing it directly is not advised
 
-julia_version = "1.8.5"
+julia_version = "1.9.3"
```
Comment (maleadt) [inline Manifest.toml:3]: It's generally safer to use the oldest supported version of Julia to generate the manifest.

**Review by maleadt — APPROVED** [review]
LGTM

File: .buildkite/pipeline.yml [hunk 2]
```diff
@@ -57,12 +57,13 @@ steps:
...
           queue: "benchmark"
           cuda: "*"
+          gpu: "rtx4070"
         if: build.message !~ /\[skip benchmarks\]/
         matrix:
           setup:
             julia:
               - "1.9"
-        timeout_in_minutes: 30
+        timeout_in_minutes: 300
```
Comment (maleadt) [inline .buildkite/pipeline.yml:66]: That's... long.

Comment (thomasfaingnaert) [reply .buildkite/pipeline.yml:66]: Ah yes, I bumped this when I got a timeout, but forgot to change it to a more reasonable number :sweat_smile:

**Pull request merged**